src/Simplex/FileTransfer/Description.hs

```diff
@@ -204,17 +229,47 @@ validateFileDescription fd@FileDescription {size, chunks}
     chunksSize = fromIntegral . foldl' (\s FileChunk {chunkSize} -> s + unFileSize chunkSize) 0
 
 encodeFileDescription :: FileDescription p -> YAMLFileDescription
-encodeFileDescription FileDescription {party, size, digest, key, nonce, chunkSize, chunks} =
+encodeFileDescription FileDescription {party, size, digest, key, nonce, chunkSize, chunks, redirect} =
   YAMLFileDescription
     { party = toFileParty party,
       size = B.unpack $ strEncode size,
       digest,
       key,
       nonce,
       chunkSize = B.unpack $ strEncode chunkSize,
-      replicas = encodeFileReplicas chunkSize chunks
+      replicas = encodeFileReplicas chunkSize chunks,
+      redirect
     }
 
+newtype FileDescriptionURI = FileDescriptionURI Text
```

Comment:
I think it should be just newtype over recipient's file description and schema:

```
data FileDescriptionURI = FileDescriptionURI ConnReqScheme (ValidFD (FileDescription 'FRecipient))
```

This way it's more useful, and can be parsed in the chat for validity.

We can rename ConnReqScheme to ServiceScheme if you want.